### DISCUSSION TITLE
Worker distribution

### DISCUSSION THREAD

**Discussion by keithhackbarth** [Kludex/uvicorn#1138] [Q&A] [answered]
I have a server setup with 12 workers. It seems that one worker receives the majority of calls. Even if it is blocked.

`gunicorn test.asgi:application -w 12 -k uvicorn.workers.UvicornWorker -b 0.0.0.0:8080 --access-logfile=-`

For more details, our endpoint has a blocking synchronous call to `requests.get`. However, I've noticed that subsequent calls to the same endpoint get routed to the same blocked worker instead of the other idle workers. So I end up seeing 1 busy worker with multiple items queued up. Does anyone have any information about how to diagnose or resolve this situation? We are using Django 3 and don't want to rewrite this specific call to be asynchronous at this moment.

**Comment by keithhackbarth** [comment, marked as answer]
Maybe related to this? https://github.com/benoitc/gunicorn/issues/1130

**Reply by euri10** [reply]
seems like a gunicorn thing indeed, on top of my head I dont see anything in uvicorn's code base where this could lead to such a thing

**Reply by keithhackbarth** [reply]
Thanks @euri10 . 

Just FYI, it seems like there is a known issue with Django/asgiref right now that everything is blocking https://github.com/django/asgiref/issues/218. And it is exacerbated by Guicorn's distribution strategy. Just leaving that info here in case anyone is dealing with the same issue as I spent a few hours thinking it was a uvicorn issue where it's actually a Django/asgiref one.

**Reply by euri10** [reply]
thanks for the heads up @keithhackbarth , that's definitely useful info if someone stumbles upon the same !
